pkg/apply/task/apply_task.go

```diff
@@ -292,6 +308,38 @@ func (a *ApplyTask) filterCRsWithCRDInSet(objects []*unstructured.Unstructured)
 	return objs, objsWithCRD, nil
 }
 
+// objInCluster returns true if the passed object is in the slice of
+// previous inventory, because an object in the previous inventory
+// exists in the cluster.
+func (a *ApplyTask) objInCluster(obj object.ObjMetadata) bool {
+	for _, clusterObj := range a.PrevInventory {
```

Comment:
I think you can change `a.PrevInventory` to a map, for example `map[object.ObjMetadata]struct{}`. Then this `for` loop can be dropped.